File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/ProjectDeploymentStatus.java
package org.kinotic.management.api.model;

/**
 * Lifecycle state of a {@link ProjectDeployment}, and why when something is wrong.
 *
 * @param type    the lifecycle state of the deployment
 * @param message why the deployment is in this state, or null when the state speaks for
 *                itself — typically the failure reason of the last deployment job
 */
public record ProjectDeploymentStatus(ProjectDeploymentStatusType type, String message) {

    public ProjectDeploymentStatus(ProjectDeploymentStatusType type) {
        this(type, null);
    }
}

File: kinotic-js/workspace/packages/management-api/src/api/model/ProjectDeploymentStatus.ts
import { ProjectDeploymentStatusType } from '@/api/model/ProjectDeploymentStatusType'

/**
 * Lifecycle state of a ProjectDeployment, and why when something is wrong.
 */
export interface ProjectDeploymentStatus {
    /**
     * The lifecycle state of the deployment.
     */
    type: ProjectDeploymentStatusType
    /**
     * Why the deployment is in this state, or null when the state speaks for itself —
     * typically the failure reason of the last deployment job.
     */
    message: string | null
}
